cognite/client/_api_client.py

```diff
@@ -136,7 +136,12 @@ def _do_request(self, method: str, url_path: str, **kwargs) -> requests.Response
         headers.update(kwargs.get("headers") or {})
 
         if json_payload:
-            data = _json.dumps(json_payload, default=utils._auxiliary.json_dump_default)
+            try:
+                data = _json.dumps(json_payload, default=utils._auxiliary.json_dump_default, allow_nan=False)
+            except ValueError as e:
```

Comment:
Just catching a bare ValueError here seems unsafe. Let's verify the error message is `Out of range float values are not JSON compliant`

Reply:
Good call, adding guard 👍 It seems that the error message is slightly longer when `_json.make_encoder` is not available:
https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Modules/_json.c#L1352 
vs
https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/json/encoder.py#L241